Sources/Core/Sources/ECS/Singles/InputState.swift

```diff
@@ -129,9 +121,11 @@ public final class InputState: SingleComponent {
     }
 
     for event in newlyReleased {
-      ///test for forwards key being released
+      // test for forwards key being released
       if event.input == .moveForward {
-        /// if the forwards key has been released, set the time since it was released
+        // if the forwards key has been released, stop sprinting
+        inputs.remove(.sprint)
```

Comment:
Sprint should only be removed when forward is released if it was triggered via double tapping. Otherwise if someone holds down their sprint key while pressing forwards, and then releases forwards and presses it again, they would end up walking even though they're still pressing the sprint key.